core/src/main/scala/com/banno/kafka/Publish.scala

```diff
@@ -16,44 +16,42 @@
 
 package com.banno.kafka
 
-import cats._
+import cats.effect._
 import cats.syntax.all._
 
 import com.banno.kafka._
 import com.banno.kafka.producer.ProducerApi
 
 import shapeless._
 
-import org.apache.avro.generic.GenericRecord
-
 object Publish {
   type T[F[_], A] = A => F[Unit]
   type KV[F[_], K, V] = T[F, (K, V)]
 
-  def to[F[_]: Functor, A, B](
+  def to[F[_]: Sync, A, B](
```

Comment:
I don't love that I had to surface `Sync` in so many places, but this was the least ridiculous solution I found that was tractable in Scala 2.